packages/headless/src/features/search/search-slice.ts

```diff
@@ -35,12 +36,18 @@ export const searchReducer = createReducer(
     builder.addCase(executeSearch.fulfilled, (state, action) => {
       handleFulfilledSearch(state, action);
       state.results = action.payload.response.results;
+      state.facetOrder = action.payload.response.facets.map(
+        (facet) => facet.facetId
+      );
     });
     builder.addCase(fetchMoreResults.fulfilled, (state, action) => {
       handleFulfilledSearch(state, action);
       state.results = [...state.results, ...action.payload.response.results];
     });
     builder.addCase(executeSearch.pending, handlePendingSearch);
     builder.addCase(fetchMoreResults.pending, handlePendingSearch);
+    builder.addCase(change.fulfilled, (state, action) => {
+      state.facetOrder = action.payload.facetOrder;
+    });
```

Comment:
More of a general comment around the fix, I think it would make sense to have the `facetOrder` property live in a dedicated slice.

If we think of tree-shaking, the facetOrder "feature" is only relevant to someone using facets. If they just have a searchbox and result list, they would need the `search` slice, but not the `facetOrder` slice.

It also feels a little strange to see history being restored in the search slice.

This more nitpick, the treeshaking benefit is not major, and what we have will work, but if it is a quick win, then I think it would be better,

Reply:
Very good point.

Major or not, it's definitely a good habit to take. I'll move the facet order elsewhere.